src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
+                self._mod_tracker[prop] = False
+
+        # Normalize types of some built-in properties
+        if prop in ('position', 'velocity', 'cameraOffset', 'positionStdDev', 'orientationStdDev'):
+            value = toVector(value, f'"{prop}" of {self} not a vector')
+        elif prop in ('width', 'length', 'visibleDistance',
+                      'viewAngle', 'speed', 'angularSpeed',
+                      'yaw', 'pitch', 'roll'):
+            value = toScalar(value, f'"{prop}" of {self} not a scalar')
+
+        if prop in ['yaw', 'pitch', 'roll']:
+            value = normalizeAngle(value)
+
+        if prop == 'regionContainedIn':
+            # 2D regions can't contain objects, so we automatically use their footprint.
+            value = convertToFootprint(value)
+
+        # Check if this property is already an attribute, unless we are overriding
+        if hasattr(self, prop) and prop not in self._propertiesSet and not overriding:
+            raise RuntimeParseError(f"Property {prop} would overwrite an attribute with the same name.")
+
+        self.properties.append(prop)
+        self._propertiesSet.add(prop)
+        object.__setattr__(self, prop, value)
+
+    def _register(self):
+        import scenic.syntax.veneer as veneer   # TODO improve?
+        veneer.registerInstance(self)
+
+    def _override(self, specifiers):
+        assert not needsSampling(self)
+        oldVals = {}
+        for spec in specifiers:
+            for prop in spec.priorities:
+                if prop in self._dynamicProperties:
+                    raise RuntimeParseError(f'cannot override dynamic property "{prop}"')
+                if prop not in self._propertiesSet:
+                    raise RuntimeParseError(f'object has no property "{prop}" to override')
+                oldVals[prop] = getattr(self, prop)
+        defs = { prop: Specifier("OverrideDefault", {prop: -1}, {prop: getattr(self, prop)}) for prop in self.properties }
+        self._applySpecifiers(list(specifiers), defs=defs, overriding=True)
+        return oldVals
+
+    def _revert(self, oldVals):
+        for prop, val in oldVals.items():
+            object.__setattr__(self, prop, val)
+
+    def sampleGiven(self, value):
+        if not needsSampling(self):
+            return self
+        return self._withProperties({ prop: value[getattr(self, prop)]
+                                    for prop in self.properties },
+                                    constProps=self._constProps)
+
+    def _allProperties(self):
+        return { prop: getattr(self, prop) for prop in self.properties }
+
+    def _copyWith(self, **overrides):
+        """Copy this object, possibly overriding some of its properties."""
+        # Copy all properties except for dynamic final values, which will retain their default values
+        props = {prop: val for prop, val in self._allProperties().items() if prop not in set(self._finalProperties)}
```

Comment:
This seems to exclude all final properties, not only dynamic ones. Is the comment wrong?